src/wp-includes/query.php

```diff
@@ -1117,7 +1116,16 @@ function _find_post_by_old_slug( $post_type ) {
 		$query .= $wpdb->prepare( ' AND DAYOFMONTH(post_date) = %d', get_query_var( 'day' ) );
 	}
 
-	$id = (int) $wpdb->get_var( $query );
+	$key          = md5( $query );
+	$last_changed = wp_cache_get_last_changed( 'posts' );
+	$cache_key    = "_find_post_by_old_slug:$key:$last_changed";
```

Comment:
Not saying this is a problem, but are there any potential caveats from using a leading underscore in cache key names? I'm just wondering about it since it strikes me as odd and I believe it doesn't exist elsewhere in core.

I understand it's because of the function name, but we could also use another related name without underscore. Just curious.